include/jwt-cpp/jwt.h

```diff
@@ -1085,6 +1089,242 @@ namespace jwt {
 			error::throw_if_error(ec);
 			return res;
 		}
+
+#if defined(JWT_OPENSSL_3_0)
+
+		/**
+		 * \brief Convert a curve name to a group name.
+		 *
+		 * \param curve	string containing curve name
+		 * \param ec	error_code for error_detection
+		 * \return 		group name
+		 */
+		inline std::string curve2group(const std::string curve, std::error_code& ec) {
+			if (curve == "P-256") {
+				return "prime256v1";
+			} else if (curve == "P-384") {
+				return "secp384r1";
+			} else if (curve == "P-521") {
+				return "secp521r1";
+			} else {
+				ec = jwt::error::ecdsa_error::unknown_curve;
+				return {};
+			}
+		}
+
+#else
+
+		/**
+		 * \brief Convert a curve name to an ID.
+		 *
+		 * \param curve	string containing curve name
+		 * \param ec	error_code for error_detection
+		 * \return 		ID
+		 */
+		inline int curve2nid(const std::string curve, std::error_code& ec) {
+			if (curve == "P-256") {
+				return NID_X9_62_prime256v1;
+			} else if (curve == "P-384") {
```

Comment:
Hi @tomhughes,

great work here 👍 

just a small suggestion for improvement. For ES256K support, the following could be added:
`
else if (curve == "secp256k1") {
    return NID_secp256k1;
}
`
I'm not totally sure if secp256k1 follows you name convention.
Maybe this is helpful for some folks.

Kind Regards & thank you!
Marcel